File: ios/NativeView.m
//
//  NativeView.m
//  mobileApp
//
//  Created by Shenu Gupta on 23/06/22.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTEventEmitter.h"
#import "React/RCTViewManager.h"

@interface RCT_EXTERN_MODULE(NativeView,RCTEventEmitter)

RCT_EXTERN_METHOD(changeToNativeView:(RCTResponseSenderBlock)callback)
RCT_EXTERN_METHOD(supportedEvents)

//RCT_EXPORT_METHOD(changeToNativeView) {
//  NSLog(@"RN binding - Native View - Loading MyViewController.swift");
//  dispatch_async(dispatch_get_main_queue(), ^{
//    AppDelegate *appDelegate = (AppDelegate *)[UIApplication sharedApplication].delegate;
//    [appDelegate goToNativeView];
//      NSLog(@"inside dispatch async block main thread from main thread");
//  });



@end

@interface RCT_EXTERN_MODULE(FirstViewController, RCTViewManager)

RCT_EXTERN_METHOD(goToNative)

@end

File: ios/NativeView.swift
//
//  NativeView.swift
//  DemoApp
//
//  Created by Shenu Gupta on 26/06/22.
//


import Foundation

@objc(NativeView)
open class NativeView: RCTEventEmitter{
  @objc public static var emitter: NativeView!
  
  override init() {
      super.init()
    NativeView.emitter = self
    }
  
  private var count = 0;
  
  @objc
  func changeToNativeView(_ callback:RCTResponseSenderBlock){
    count += 1;
//    print(count);
    callback([count])
    DispatchQueue.main.async {
      let appDelegate = UIApplication.shared.delegate as! AppDelegate
      appDelegate.goToNativeView()
     // self.sendEvent(withName: "EventReminder", body: ["count increase",self.count])
      }
  }
  
  @objc
  public override static func requiresMainQueueSetup() ->Bool{
    return true;
  }
  
  @objc
  open override func constantsToExport() -> [AnyHashable: Any]!{
    return ["initialCount": 0];
  }
  
  open override func supportedEvents() -> [String]! {
    return ["onIncrement","onDecrement","EventReminder"];
  }
  @objc
  func callJSCallbacks() {
    let appDelegate = UIApplication.shared.delegate as! AppDelegate
    self.sendEvent(withName: "EventReminder", body: ["count increase",self.count])
  }
}

